torch/utils/data/_utils/pin_memory.py

```diff
@@ -52,6 +52,10 @@ def pin_memory_batch(batch):
     elif isinstance(batch, container_abcs.Mapping):
         return {k: pin_memory_batch(sample) for k, sample in batch.items()}
     elif isinstance(batch, container_abcs.Sequence):
-        return [pin_memory_batch(sample) for sample in batch]
+        batch_list = [pin_memory_batch(sample) for sample in batch]
+        try:  # e.g. if namedtuple
+            return type(batch)(*batch_list)
+        except TypeError:
+            return type(batch)(batch_list)
```

Comment:
Honestly this code makes me very uneasy, and it favors the less-common case of using a namedtuple by adding overhead to code that works today. I think an `if isinstance(..., NamedTuple)` would be better.

Reply:
I understand. So then I will just squeeze in an extra `elif isinstance(...)` on line 54 of the original file.

Reply:
Turns out there is no clean way to check for instances of namedtuples. I can think of three alternatives:

1. Change order of `try` and `except` to favor common classes and reduce overhead. But I guess that namedtuples with only 1 field would mistake the batch_list for its element, and not throw the due exception.
2. Insert an `elif isinstance(batch, tuple) and hasattr(batch, '_fields'):` on line 54. This is the suggested Duck typing solution from https://bugs.python.org/issue7796
3. Drop the patch altogether because namedtuples are too awkward.